micropip/wheelinfo.py

```diff
@@ -69,8 +69,15 @@ def from_url(cls, url: str) -> "WheelInfo":
         See https://www.python.org/dev/peps/pep-0427/#file-name-convention
         """
         parsed_url = urlparse(url)
-        if parsed_url.scheme == "":
-            url = "file:///" + url
+        if not parsed_url.scheme:
+            parsed_url = ParseResult(
+                scheme="file",
+                netloc=parsed_url.netloc,
+                path=parsed_url.path,
+                params=parsed_url.params,
+                query=parsed_url.query,
+                fragment=parsed_url.fragment,
+            )
```

Comment:
I think we need to keep the `url = "file:///" + url` as well. 

How about:

```python
if parsed_url.scheme == "":
    url = "file:///" + url
    parsed_url = urlparse(url)
```

?

Reply:
🤔 I think this is handled by `if not parsed_url.scheme`, right? Could you please clarify?

Reply:
Ah, I did try this in this commit: https://github.com/pyodide/micropip/pull/183/commits/09772c810b3529751b5b1168169c5bcac59d1758, but it didn't seem to respect the URL scheme even when passing `scheme='file'` directly, so I reverted it.

Reply:
Oh, I did try this in this commit: https://github.com/pyodide/micropip/pull/183/commits/09772c810b3529751b5b1168169c5bcac59d1758, but it didn't seem to respect the URL scheme even when passing `scheme='file'` directly and the tests failed, so I had to revert it. Did I miss anything?

Reply:
Hmm, got it. That's weird. Thanks for checking! Probably we can improve how `url` and `parsed_url` are managed inside the `WheelInfo` class, but we can do that later.